crates/fluvio-spu/src/services/public/stream_fetch.rs

```diff
@@ -130,6 +134,25 @@ impl StreamFetchHandler {
         let max_bytes = msg.max_bytes as u32;
         let sm_engine = ctx.smartstream_owned();
 
+        let maybe_join = if let Some(SmartStreamKind::Join(topic_name)) =
+            msg.smart_module.as_ref().map(|wasm| &wasm.kind)
+        {
+            let consumer = ctx.leaders().partition_consumer(topic_name, 0).await;
+
+            let join_stream = match consumer.stream(fluvio::Offset::beginning()).await {
+                Ok(stream) => stream,
+                Err(err) => {
+                    error!("error fetching join data {}", err);
```

Comment:
@sehz I did the update,

This is throwing this log:

```sh
2021-11-08T03:49:46.609005Z ERROR fetch{stream_id=3 replica=left-0 sink=26}: fluvio_spu::services::public::stream_fetch: error fetching join data Fluvio socket error

```

Reply:
I test this with:

```
fluvio topic create left
fluvio topic create right
cd crates/fluvio-smarstream/examples && cargo build --release -p fluvio-wasm-join
fluvio consume left --join target/wasm32-unknown-unknown/release/fluvio_wasm_join.wasm --join-topic right

### And in another shell produce numbers to right and left topic (we should get the sum of the last record in right with the records in left)

Reply:
Thanks.  I will try it out

Reply:
Works